compiler/src/iree/compiler/Codegen/Common/test/iree_comprehensive_bufferize.mlir

```diff
@@ -2571,3 +2571,21 @@ func.func @sub_byte_bufferize_with_offset() {
 //       CHECK:   %[[C64:.+]] = arith.constant 64 : index
 //       CHECK:   hal.interface.binding.subspan set(0) binding(0)
 //  CHECK-SAME:       memref<64xi4, strided<[1], offset: 128>
+
+// -----
+
+func.func @tensor_barrier() -> vector<2xf32> {
+  %cst = arith.constant dense<0.0> : vector<2xf32>
+  %cst0 = arith.constant 0.0 : f32
+  %c0 = arith.constant 0 : index
+  %alloc = bufferization.alloc_tensor() : tensor<2xf32>
+  %tmp = vector.transfer_write %cst, %alloc[%c0] {in_bounds = [true]} : vector<2xf32>, tensor<2xf32>
+  %barrier = iree_gpu.tensor_barrier %tmp : tensor<2xf32>
+  %res = vector.transfer_read %barrier[%c0], %cst0 {in_bounds = [true]} : tensor<2xf32>, vector<2xf32>
+  return %res : vector<2xf32>
+}
```

Comment:
Can you add a test that goes through a loop? I'm curious what happens there.

Reply:
By loop do you mean a barrier on the iterator argument?

Reply:
added